client_plugin/drivers/shared/shared_driver.go

```diff
@@ -91,8 +174,47 @@ func (d *VolumeDriver) List(r volume.Request) volume.Response {
 // GetVolume - return volume meta-data.
 func (d *VolumeDriver) GetVolume(name string) (map[string]interface{}, error) {
 	var statusMap map[string]interface{}
+	var volRecord VolumeMetadata
 	statusMap = make(map[string]interface{})
-	log.Errorf("VolumeDriver GetVolume to be implemented")
+
+	// The kv pairs we want from the KV store
+	keys := []string{
+		etcdPrefixState + name,
+		etcdPrefixGRef + name,
+		etcdPrefixInfo + name,
+	}
+
+	// KV pairs will be returned in same order in which they were requested
+	entries, err := d.readVolMetadata(keys)
+	if err != nil {
+		msg := fmt.Sprintf("Failed to read volume metadata from etcd. %v", err)
+		log.Warningf(msg)
+		return statusMap, errors.New(msg)
+	}
+	if len(entries) == 0 {
+		// Volume does not exist!
+		msg := fmt.Sprintf("Volume does not exist")
+		return statusMap, errors.New(msg)
+	} else if len(entries) != len(keys) {
+		// Volume exists but we didn't find all its keys
+		msg := fmt.Sprintf("Failed to get volume: Couldn't find all keys!")
```

Comment:
Logically this should not happen. We create and delete all three type of keys at the same time.
So when this situation happens, I cannot think of a reason other than data corruption for now.
And I am not sure if etcd can have data corruption?